homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Sensors for GoDice device."""
+
+import logging
+
+import godice
+
+from homeassistant import const
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_DEVICE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ROLLED_NUMBER_SENSOR_DESCR = SensorEntityDescription(
+    key="rolled_number", state_class=SensorStateClass.MEASUREMENT
+)
+COLOR_SENSOR_DESCR = SensorEntityDescription(
+    key="color",
+    device_class=SensorDeviceClass.ENUM,
+    options=[color.name for color in godice.Color],
+)
+BATTERY_SENSOR_DESCR = SensorEntityDescription(
+    key="battery_level",
+    device_class=SensorDeviceClass.BATTERY,
+    native_unit_of_measurement=const.PERCENTAGE,
+    state_class=SensorStateClass.MEASUREMENT,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Setups Dice sensors."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    device = data[DATA_DEVICE]
+    device_name = config_entry.data[const.CONF_NAME]
+    device_info = DeviceInfo(
+        identifiers={(DOMAIN, device_name)},
+        name=device_name,
+        manufacturer="Particula",
+        model="GoDice",
+    )
+
+    sensor_create_data = [
+        (DiceColorSensor, COLOR_SENSOR_DESCR),
+        (BatteryLevelSensor, BATTERY_SENSOR_DESCR),
+        (DiceNumberSensor, ROLLED_NUMBER_SENSOR_DESCR),
+    ]
+
+    async_add_entities(
+        [
+            entity_class(device_info, entity_description, device)
+            for entity_class, entity_description in sensor_create_data
+        ]
+    )
+
+
+class BaseSensor(SensorEntity):
+    """Base for concrete sensors."""
+
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+
+    dice: godice.Dice
+
+    def __init__(
+        self,
+        device_info: DeviceInfo,
+        entity_description: SensorEntityDescription,
+        device: godice.Dice,
+    ) -> None:
+        """Set default values."""
+        self.entity_description = entity_description
+        self.dice = device
+        self._attr_device_info = device_info
+        self._attr_unique_id = (
+            f"{device_info[const.CONF_NAME]}_{self.entity_description.key}"
+        )
+        self._attr_translation_key = entity_description.key
```

Comment:
Is there a reason you write `self.dice = device` and not `self.dice = dice`? Then you can use `device` for HA device related things. In that case I would also change `DATA_DEVICE` to `DATA_DICE` as that would make it even more clear in my opinion, meaning all things mentioning `device` are related to HA while `dice` is the godice

```python
dice = hass.data[DOMAIN][config_entry.entry_id][DATA_DICE]
```